stdlib/in_channel.mli

```diff
@@ -148,6 +148,25 @@ val input_all : t -> string
     If the same channel is read concurrently by multiple threads, the returned
     string is not guaranteed to contain contiguous characters from the input. *)
 
+val input_lines : t -> string list
+(** [input_lines ic] reads lines using {!input_line}
+    until the end of file is reached.  It returns the list of all
+    lines read, in the order they were read.  The newline characters
+    that terminate lines are not included in the returned strings.
+    Empty lines produce empty strings.
+
+    @since 5.1 *)
+
+val fold_lines : ('a -> string -> 'a) -> 'a -> t -> 'a
+(** [fold_lines f init c] reads lines from [ic] using {!input_line}
+    until the end of file is reached.  Each line is passed to function [f]
+    in the style of a fold.  More precisely, [fold_lines f init c] is
+    [List.fold_left f init (In_channel.input_lines ic)],
```

Comment:
```suggestion
    in the style of a fold.  More precisely, [fold_lines f init ic] is
    [List.fold_left f init (In_channel.input_lines ic)],
```

More precisely, though no more accurately. The interleaving of side effects is different: `fold_lines` works on infinite streams and allows writing (a contrived form of) interactive programs.

This could be documented using similar language as `Seq.fold_left`.